### PR TITLE
EVG-3014: Adding patch-only tasks

### PR DESCRIPTION
**Pull request by ybrill**

### CODE REVIEW

File: model/project_parser_test.go [hunk 1]
```diff
@@ -1115,3 +1117,44 @@ buildvariants:
 	assert.Equal("task_3", proj.BuildVariants[2].Tasks[0].Requires[0].Name)
 	assert.Equal("task_3", proj.BuildVariants[2].Tasks[1].Requires[0].Name)
 }
+
+func TestPatchOnlyTasks(t *testing.T) {
+	assert := assert.New(t)
+	yml := `
+tasks:
+- name: task_1
+  patch_only: true
```
[the remark this answers is not in the document]
Comment (ybrill) [reply model/project_parser_test.go]: - What are you saying looks like it won't work?
- As I understand, you can specify it on the task itself and/or on the bvt override. On 1126 it's on the task and on 1133 it's on the bvt. The overriding takes place in project.go:127. What am I missing?

File: model/lifecycle.go [hunk 2]
```diff
@@ -605,14 +605,16 @@ func CreateTasksFromGroup(in BuildVariantTaskUnit, proj *Project) []BuildVariant
...
+func isPatchBuild(requester string) bool {
+	return evergreen.IsPatchRequester(requester)
+}
+
+func skipOnPatchBuild(t BuildVariantTaskUnit) bool {
+	return (t.Patchable != nil && !*t.Patchable) || t.Name == evergreen.PushStage
+}
+
+func skipOnNonPatchBuild(patchOnly *bool) bool {
+	return patchOnly != nil && *patchOnly
```
[the remark this answers is not in the document]
Comment (ybrill) [reply model/lifecycle.go]: Should be addressed now.

File: model/project_parser_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ybrill) [reply model/project_parser_test.go]: It looks to me that it should be supported in tasks as well:
- the value is parsed from the config file into project_parser.go:100 (or line 71 for a task group). These are part of the parserProject struct (line 60)
- translateProject (line 447) parses the parserTasks into ProjectTasks (with evaluateTaskUnits).
- The value of the project task is adopted when there is no value for the bv task in project.go:127

**Comment by tychoish** [comment]
I agree with john that we should check this in staging just to be sure.

**Pull request merged**